src/ZipkinBundle/Components/Messenger/ZipkinSendHandler.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+
+namespace ZipkinBundle\Components\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+
+use Zipkin\Kind;
+use Zipkin\Propagation\B3;
+use Zipkin\Tags;
+use Zipkin\Tracing;
+
+class ZipkinSendHandler implements ZipkinHandlerInterface
+{
+    /**
+     * @var \Zipkin\Tracer
+     */
+    private $tracer;
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracer = $tracing->getTracer();
+    }
+
+    public function handle(Envelope $envelope): Envelope
+    {
+        if ($envelope->all(ZipkinStamp::class)) {
+            return $envelope;
+        }
+
+        $span = $this->tracer->nextSpan();
+        $span->start();
+
+        $span->setKind(Kind\PRODUCER);
+        $span->tag(Tags\LOCAL_COMPONENT, 'symfony');
+        $span->setName(get_class($envelope->getMessage()));
+
+        $stamp = new ZipkinStamp;
+        $stamp->add(B3::SPAN_ID_NAME, $span->getContext()->getSpanId());
```

Comment:
This one is interesting. `B3::SPAN_ID_NAME`'s value is `x-b3-spanid` but that is very http specific, what do we do in messaging? @jeqo @adriancole

Reply:
if you mean propagation, we do single-header propagation https://github.com/openzipkin/b3-propagation#single-header

Reply:
Awesome, I started this: https://github.com/openzipkin/zipkin-php/pull/151

Reply:
@jcchavezs done. Could you check if you like it?